docs/user-guide/grpc.md

```diff
@@ -0,0 +1,136 @@
+# gRPC example
+
+This section explains how to use Traefik has reverse proxy for gRPC application with self-signed certificates.
+
+!!! warning
+    Has gRPC needs HTTP2, we need to have valid HTTPS on both gRPC Server and Traefik.
+
+<p align="center">
+<img src="/img/grpc.svg" alt="gRPC architecture" title="gRPC architecture" />
+</p>
+
+## gRPC Server certificate
+In order to secure the gRPC server, we generate a self-signed certificate for backend url.
+```bash
+ openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout ./backend.key -out ./backend.crt
+```
+
+That will prompt for information, the important answer is
+```
+Common Name (e.g. server FQDN or YOUR name) []: backend.local
+```
```

Comment:
Maybe show the command line with all information to key (with fake values) and add a note to explain the importance of the `CN` value?

```bash
# Set our CSR variables
SUBJ="
C=FR
ST=MP
O=
localityName=Toulouse
commonName=$DOMAIN
organizationalUnitName=Traefik
emailAddress=
"
 openssl req -x509 -nodes -days 365 -newkey rsa:2048 -subj "$(echo -n "$SUBJ" | tr "\n" "/")" -keyout ./backend.key -out ./backend.crt
```

(command to  check)